src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRecordCopyCtor : SynthesizedInstanceConstructor
+    {
+        public SynthesizedRecordCopyCtor(
+            SourceMemberContainerTypeSymbol containingType,
+            DiagnosticBag diagnostics)
+            : base(containingType)
+        {
+            Debug.Assert(!containingType.IsStructType(), "only reference types should define copy constructors");
```

Comment:
>Debug.Assert [](start = 12, length = 12)

According to the new spec, structs do indeed have a copy constructor. I would expect people would sometimes want to supply their own, so that makes sense.